### PR TITLE
chore: configure unit tests for ci cd setup

### PR DESCRIPTION
**Pull request by colinscz**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Build-related changes
- [x] Other, please describe:
- [x] Remove E2e tests which are outdated and not working

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `main` branch for v2.x (or to a previous version branch)
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] All tests are passing
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: libraries/ui-library/src/components/six-form/test/six-form.spec.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { newSpecPage } from '@stencil/core/testing';
 import { SixForm } from '../six-form';
 
 describe('six-form', () => {
-  it('renders', async () => {
+  xit('renders', async () => {
```
Comment (Riatomerehentroweard) [inline libraries/ui-library/src/components/six-form/test/six-form.spec.tsx:5]: Why is this test muted? I'm not a fan of bringing dead code into the main branch. Is the test currently broken?

File: libraries/ui-library/src/components/six-header/test/six-header.spec.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { newSpecPage } from '@stencil/core/testing';
 import { SixHeader } from '../six-header';
 
 describe('six-header', () => {
-  it('renders', async () => {
+  xit('renders', async () => {
```
Comment (Riatomerehentroweard) [inline libraries/ui-library/src/components/six-header/test/six-header.spec.tsx:5]: Same here why is this test muted? I'm not a fan of bringing dead code into the main branch. Is the test currently broken?

File: libraries/ui-library/src/components/six-header/test/six-header.spec.tsx [hunk 3]
```diff
@@ -67,7 +67,7 @@ describe('six-header', () => {
     `);
   });
 
-  it('should render app switcher', async () => {
+  xit('should render app switcher', async () => {
```
Comment (Riatomerehentroweard) [inline libraries/ui-library/src/components/six-header/test/six-header.spec.tsx:70]: Hm same here why is this test muted? I'm not a fan of bringing dead code into the main branch. Is the test currently broken?

**Review by Riatomerehentroweard — CHANGES_REQUESTED** [review]
The build seems to be still failing and I would not submit muted tests.

File: libraries/ui-library/src/components/six-header/test/six-header.spec.tsx (hunk 3 above)
Comment (colinscz) [reply libraries/ui-library/src/components/six-header/test/six-header.spec.tsx:70]: This test is currently broken, that is why I deactivated it for now. 

Do you want me to remove it instead? Or should we dedicate specific time to get those unit tests running. Then I would address this on a separate issue as the goal of this PR was to introduce the existing tests without modificaiton.

**Comment by colinscz** [comment]
> The build seems to be still failing and I would not submit muted tests.

@Riatomerehentroweard build is now fixed

File: libraries/ui-library/src/components/six-header/test/six-header.spec.tsx (hunk 3 above)
Comment (colinscz) [reply libraries/ui-library/src/components/six-header/test/six-header.spec.tsx]: were adressed

File: libraries/ui-library/src/components/six-form/test/six-form.spec.tsx (hunk 1 above)
Comment (colinscz) [reply libraries/ui-library/src/components/six-form/test/six-form.spec.tsx]: were adressed

**Review by Riatomerehentroweard — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -24,6 +24,7 @@ jobs:
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
       - run: npm ci
+      - run: npm run test:unit
       - run: npm run prettier:check
```
Comment (ibirrer) [inline .github/workflows/ci.yml:28]: not really part of this PR, but we should probably move the `prettier:check` one line down to make sure the code is still formatted correctly after the build.

File: libraries/ui-library/package.json [hunk 5]
```diff
@@ -45,18 +45,17 @@
...
     "eslint-config-prettier": "^8.3.0",
     "eslint-plugin-prettier": "^4.0.0",
     "eslint-plugin-react": "^7.28.0",
-    "jest": "^26.6.3",
-    "jest-cli": "^26.6.3",
+    "jest": "27.4.5",
+    "jest-cli": "27.4.5",
     "material-icons": "^1.12.0",
-    "puppeteer": "^13.3.2",
-    "puppeteer-core": "^13.3.2",
```
Comment (ibirrer) [inline libraries/ui-library/package.json:59]: Can't we remove puppeteer completely, now that all e2e-tests are deleted?

**Comment by ibirrer** [comment]
Do we still need the screenshots or could we delete those as well?

File: package.json [hunk 6]
```diff
@@ -4,6 +4,7 @@
   "scripts": {
     "build": "npm run build --workspaces",
     "build:lib": "npm run build -w libraries",
+    "test:unit": "npm run test -w libraries/ui-library",
```
Comment (ibirrer) [inline package.json:7]: How about replacing this with `npm run test --workspaces --if-present`, which would run all tests for all libraries and examples if any these have a test script defined. This would make sure that all tests are run if we start to write tests for other packages.

**Review by ibirrer — DISMISSED** [review]

File: libraries/ui-library/package.json [hunk 7]
```diff
@@ -45,18 +45,17 @@
...
+    "puppeteer": "^20.2.1",
```
Comment (ibirrer) [inline libraries/ui-library/package.json:58]: Can't we remove puppeteer completely, now that all e2e-tests are deleted?

Comment (colinscz) [reply libraries/ui-library/package.json:58]: No we can't. I tried and if you check one of the last runs these fail then since the stencil-test package somehow underneath references puppeteer. It's annoying but we have to keep it for now due to that.

File: package.json (hunk 6 above)
Comment (colinscz) [reply package.json:7]: Since the other libs have the test setup not properly configured they fail the build if I add this change. I would suggest we add this once we setup the testing  configuration for the other libraries.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (colinscz) [reply .github/workflows/ci.yml]: Moved it down 35b632c99f371df0e6fd50040aaf13b9c785a5da

**Review by ibirrer — APPROVED** [review]

**Comment by colinscz** [comment]
> Do we still need the screenshots or could we delete those as well?

I don't think we need them anymore, good catch - I deleted the whole folder for now

**Review by ibirrer — APPROVED** [review]
